### PR TITLE
New-AzureRmSqlDatabase -SampleName

### PR DESCRIPTION
**Pull request by jaredmoo**
## Description
Add -SampleName param to New-AzureRmSqlDatabase.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Sql/AzureRM.Sql.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '2.8.0'
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/AzureRM.Sql.psd1]: @jaredmoo please revert this change. We will bump the version during the release

File: src/ResourceManager/Sql/ChangeLog.md [hunk 2]
```diff
@@ -19,11 +19,15 @@
 -->
 ## Current Release
 
+## Version 2.9.0
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/ChangeLog.md]: @jaredmoo please remove this "Version 2.9.0" header you have added and place additions to the change log under "Current Release"

**Comment by jaredmoo** [comment]
Build error: http://azuresdkci.cloudapp.net/job/powershell/8122/artifact/src/Package/HelpIssues.csv/*view*/

File: src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/NewAzureSqlDatabase.cs [hunk 3]
```diff
@@ -101,6 +100,11 @@ public class NewAzureSqlDatabase : AzureSqlDatabaseCmdletBase
         [Alias("Tag")]
         public Hashtable Tags { get; set; }
 
+        [Parameter(Mandatory = false,
+            HelpMessage = "The name of the sample schema to apply when creating this database.")]
+        [ValidateNotNullOrEmpty]
+        public string SampleName { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/NewAzureSqlDatabase.cs]: @jaredmoo would you mind providing an example of when a user would use this parameter? Also, how does the user discover which sample schemas they may provide to this cmdlet? I just want to make sure I understand the scenario trying to be covered by this addition and if `ValidateSet` should be used (if there is a predetermined set of schemas that can be used)

Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/NewAzureSqlDatabase.cs]: This parameter is used to create a database populated with some sample tables, data, etc. The only allowed sample name for now is `AdventureWorksLT` which is one of the sample db's we publish for SQL server. See https://msftdbprodsamples.codeplex.com/ . However in theory there could be other samples in the future.

Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/NewAzureSqlDatabase.cs]: @jaredmoo do you think we should add the `ValidateSet` attribute to this parameter so that only specific values can be passed through? You can always add more values later when more samples are created. This will prevent users from putting in `Foo` and getting an error saying that the sample `Foo` does not exist and allow them to tab through the possible samples that are offered.

Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/NewAzureSqlDatabase.cs]: Sure. Done

**Comment by cormacpayne** [comment]
@jaredmoo Hey Jared, would you mind pulling the latest changes from the `preview` branch? These changes removed the dll-Help.xml file from the repository, so you will no longer need to update it in this PR.

**Comment by jaredmoo** [comment]
@cormacpayne done thx

File: src/ResourceManager/Sql/Commands.Sql/help/New-AzureRmSqlDatabase.md [hunk 4]
```diff
@@ -163,13 +163,42 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+<<<<<<< HEAD
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/help/New-AzureRmSqlDatabase.md]: @jaredmoo looks like this file didn't survive the merge conflict

Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql/help/New-AzureRmSqlDatabase.md]: Oops, thanks for spotting this.
